### ISSUE TITLE
Isolate runtime deps

### ISSUE THREAD

**Issue by bbatsov** [xsc/lein-ancient#56]
lein-ancient's dependencies might interfere with a project's own dependencies. It'd best to isolate its runtime dependencies using a tool like https://github.com/benedekfazekas/mranderson or https://github.com/jafingerhut/dolly

**Comment by xsc** [comment]
Those look interesting. I'll have a look.
Thanks!

**Comment by bbatsov** [comment]
Btw, here's the PR that got this integrated into `cider-nrepl` https://github.com/clojure-emacs/cider-nrepl/pull/183

Might be useful as well.

**Comment by aiba** [comment]
I think I'm running into this issue.

With `:pedantic? :abort` and plugins that include both `[lein-ancient "0.6.9-SNAPSHOT"]` and `[lein-figwheel "0.5.2"]`, I get the error:

```
Possibly confusing dependencies found:
[lein-ancient "0.6.9-SNAPSHOT"] -> [org.clojure/tools.reader "0.10.0"]
 overrides
[lein-figwheel "0.5.2"] -> [figwheel "0.5.2"] -> [org.clojure/clojurescript "1.7.228" :exclusions [org.apache.ant/ant]] -> [org.clojure/tools.reader "1.0.0-alpha1"]
 and
[lein-figwheel "0.5.2"] -> [figwheel-sidecar "0.5.2"] -> [org.clojure/clojurescript "1.7.228" :exclusions [org.apache.ant/ant]] -> [org.clojure/tools.reader "1.0.0-alpha1"]

Consider using these exclusions:
[lein-figwheel "0.5.2" :exclusions [org.clojure/tools.reader]]
[lein-figwheel "0.5.2" :exclusions [org.clojure/tools.reader]]
```

My workaround is to put into `~/.lein/profiles.clj`:

```
:ancient {:pedantic? true ;; warnings but don't abort
          :plugins [[lein-ancient "0.6.9-SNAPSHOT"]]}
```

And run ancient with `lein with-profile +ancient ancient`.

This also has the side benefit of not loading `lein-ancient`'s deps when doing normal development/production lein tasks.

**Comment by xsc** [comment]
Dependency isolation has landed in 0.6.10.

[Release Notes](https://github.com/xsc/lein-ancient/releases/tag/clojars-0.6.10)
